pkg/network/dhcp/dhcp.go

```diff
@@ -38,6 +58,62 @@ func (d *Probe) Start(inf string) error {
 	return nil
 }
 
+// Start runs the DHCP client process and registers it to zinit
+func (d *BackgroundProbe) Start(inf string) error {
```

Comment:
I'm not a big fan of how this is implemented. The tricks to detect in which network namespace we are etc seems error prone.
Why can't we just specify the network namespace if there is any in the NewBackgroundProbe function ? The `inf` argument should also be on the `NewBackgroundProbe` instead of the `Start` method. The idea is you create your object with all it's property and then call `Start`/`Stop` on it without any further argument.